### PR TITLE
Document server.toml

### PR DESCRIPTION
**Pull request by jakkaz**
Updated the docs inside the server.toml file accordingly to https://docs.iggy.rs/server/configuration and sometimes guesses/suspicions (please validate them) 🙂 
This closes the issue #260.

### CODE REVIEW

File: configs/server.toml [hunk 1]
```diff
@@ -1,122 +1,337 @@
...
+
+# Maximum size of the cache, e.g., "4GB".
 size = "4GB"
 
+# Data retention policy configuration.
 [system.retention_policy]
+# Configures the message expiry setting.
+# "disabled" means messages are kept indefinitely.
+# A time value in seconds determines the lifespan of messages.
 message_expiry = "disabled"
```
Comment (hubcio) [inline configs/server.toml:259]: please mention that example format, eg. 2 days 4 hours 15 minutes

File: configs/server.toml [hunk 2]
```diff
@@ -1,122 +1,337 @@
...
+
+# The threshold of buffered messages before triggering a save to disk (integer).
+# Specifies how many messages accumulate before persisting to storage.
+# Adjusting this can balance between write performance and data durability.
 messages_required_to_save = 10_000
 
+# Segment configuration
 [system.segment]
+# Defines the maximum size of a storage segment in bytes (integer).
+# When a segment reaches this size, a new segment is created for subsequent data.
```
Comment (hubcio) [inline configs/server.toml:328]: please mention that it's soft limit, maximum segment size is actually this value + sizeof single message

**Comment by hubcio** [comment]
btw: have you already joined our discord?

File: configs/server.toml [hunk 3]
```diff
@@ -1,122 +1,337 @@
...
 interval = 60
 
+# System configuration.
 [system]
+# Base path for system data storage.
 path = "local_data"
 
+# Database configuration.
 [system.database]
+# Path for storing database files.
```
Comment (spetz) [inline configs/server.toml:225]: It might be worth mentioning that it's relative to `system.path`, other than that everything else looks good :)

**Review by spetz — CHANGES_REQUESTED** [review]

File: configs/server.toml (hunk 3 above)
Comment (jakkaz) [reply configs/server.toml:225]: Updated

File: configs/server.toml (hunk 1 above)
Comment (jakkaz) [reply configs/server.toml:259]: Updated, please check if this example is good enough

File: configs/server.toml (hunk 2 above)
Comment (jakkaz) [reply configs/server.toml:328]: Updated, I also put the explanation behind in a form of example

File: configs/server.toml (hunk 1 above)
Comment (jakkaz) [reply configs/server.toml:259]: I saw this one https://github.com/iggy-rs/iggy/issues/277 mentioning about human readable values for configs. Would you like to already introduce it in this PR as an example? If yes, I can update it to the example proposed by you

Comment (spetz) [reply configs/server.toml:259]: @jakkaz we're using now a human-friendly strings for parsing the time (docs site needs to be updated with this info too).

Comment (jakkaz) [reply configs/server.toml:259]: gotcha, updated then to the human readable format together with example

**Review by spetz — APPROVED** [review]

**Review by hubcio — APPROVED** [review]

**Pull request merged**